src/citrine/informatics/constraints/ingredient_ratio_constraint.py

```diff
@@ -95,61 +95,54 @@ def label(self, value: Optional[Tuple[str, float]]):
         self._labels = self._numerator_validate(value, "Label")
 
     @property
-    def basis_ingredients(self) -> Mapping[str, float]:
+    def basis_ingredients(self) -> Set[str]:
         """Retrieve the ingredients in the denominator of the ratio."""
-        warnings.warn("basis_ingredients is deprecated as of 2.13.0 and will change in 3.0. "
-                      "Please use basis_ingredient_names instead.", DeprecationWarning)
-        return self._basis_ingredients
+        return set(self._basis_ingredients.keys())
 
     @basis_ingredients.setter
-    def basis_ingredients(self, value: Union[Set[str], Mapping[str, float]]):
+    def basis_ingredients(self, value: Set[str]):
         """Set the ingredients in the denominator of the ratio."""
-        if isinstance(value, dict):
-            warnings.warn("As of version 2.13.0, multipliers for all basis ingredients are "
-                          "ignored, so basis_ingredients should be a list of ingredient names.",
-                          DeprecationWarning)
-            value = set(value.keys())
-
-        self.basis_ingredient_names = value
+        self._basis_ingredients = dict.fromkeys(value, 1)
 
     @property
     def basis_ingredient_names(self) -> Set[str]:
         """Retrieve the names of all ingredients in the denominator of the ratio."""
-        return set(self._basis_ingredients.keys())
+        warnings.warn("basis_ingredient_names is deprecated as of 3.0.0 and will be dropped in "
+                      "4.0. Please use basis_ingredients instead.", DeprecationWarning)
+        return self.basis_ingredients
 
     # This is for symmetry; it's not strictly necessary.
     @basis_ingredient_names.setter
     def basis_ingredient_names(self, value: Set[str]):
         """Set the names of all ingredients in the denominator of the ratio."""
-        self._basis_ingredients = dict.fromkeys(value, 1)
+        warnings.warn("basis_ingredient_names is deprecated as of 3.0.0 and will be dropped in "
+                      "4.0. Please use basis_ingredients instead.", DeprecationWarning)
```

Comment:
Why aren't we deprecating this prior to the 3.0 release, and just deleting then?

Reply:
https://citrine.atlassian.net/browse/PLA-12471

Reply:
Short version: changed how an existing method worked, this was to allow migrating before moving to 3.0.